From 3b069f16a2e4aedeb68051708c182e42975a55e4 Mon Sep 17 00:00:00 2001
From: Liming Wang <liming.wang@windriver.com>
Date: Tue, 25 Mar 2014 18:40:22 +0800
Subject: [PATCH 177/628] char: xilinx_devcfg: Remove __devinit, __devexit,
 __definitdata, __devexit_p, __exit_p

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 068160c0 Merge commit 'v3.8' into master-next

cherry picked the xilinx devcfg driver modification.

Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/char/xilinx_devcfg.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/char/xilinx_devcfg.c b/drivers/char/xilinx_devcfg.c
index 58c1df3..4f50fff 100644
--- a/drivers/char/xilinx_devcfg.c
+++ b/drivers/char/xilinx_devcfg.c
@@ -1513,7 +1513,7 @@ static const struct attribute_group xdevcfg_attr_group = {
  *
  * It does all the memory allocation and registration for the device.
  */
-static int __devinit xdevcfg_drv_probe(struct platform_device *pdev)
+static int xdevcfg_drv_probe(struct platform_device *pdev)
 {
 	struct resource *regs_res, *irq_res;
 	struct xdevcfg_drvdata *drvdata;
@@ -1688,7 +1688,7 @@ failed0:
  *
  * Unregister the device after releasing the resources.
  */
-static int __devexit xdevcfg_drv_remove(struct platform_device *pdev)
+static int xdevcfg_drv_remove(struct platform_device *pdev)
 {
 	struct xdevcfg_drvdata *drvdata;
 	struct resource *res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
@@ -1715,7 +1715,7 @@ static int __devexit xdevcfg_drv_remove(struct platform_device *pdev)
 	return 0;		/* Success */
 }
 
-static struct of_device_id xdevcfg_of_match[] __devinitdata = {
+static struct of_device_id xdevcfg_of_match[] = {
 	{ .compatible = "xlnx,ps7-dev-cfg-1.00.a", },
 	{ /* end of table */}
 };
@@ -1724,7 +1724,7 @@ MODULE_DEVICE_TABLE(of, xdevcfg_of_match);
 /* Driver Structure */
 static struct platform_driver xdevcfg_platform_driver = {
 	.probe = xdevcfg_drv_probe,
-	.remove = __devexit_p(xdevcfg_drv_remove),
+	.remove = xdevcfg_drv_remove,
 	.driver = {
 		.owner = THIS_MODULE,
 		.name = DRIVER_NAME,
-- 
1.7.5.4

